File: CS/ViewModel/ChildPOCOViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;

namespace Example.ViewModel {
    public class ChildPOCOViewModel : ISupportParameter {
        [ServiceProperty(SearchMode=ServiceSearchMode.PreferParents)]
        protected virtual IMessageBoxService MessageBoxService { get { return null; } }
        public virtual object Parameter { get; set; }

        protected virtual void OnParameterChanged() {
            if(Parameter is string)
                MessageBoxService.Show("ChildPOCOViewModel: Parameter = " + (string)Parameter);
        }
    }
}

File: VB/ViewModel/ChildPOCOViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations

Namespace Example.ViewModel

    Public Class ChildPOCOViewModel
        Implements ISupportParameter

        <ServiceProperty(SearchMode:=ServiceSearchMode.PreferParents)>
        Protected Overridable ReadOnly Property MessageBoxService As IMessageBoxService
            Get
                Return Nothing
            End Get
        End Property

        Public Overridable Property Parameter As Object Implements ISupportParameter.Parameter

        Protected Overridable Sub OnParameterChanged()
            If TypeOf Parameter Is String Then MessageBoxService.Show("ChildPOCOViewModel: Parameter = " & CStr(Parameter))
        End Sub
    End Class
End Namespace
